### PR TITLE
RNMT-4515 Add support for Android 11

### PR DESCRIPTION
**Pull request by luissilvaos**
### Description
This PR introduces an hotfix to ensure the support for Android 11.

The current version of the plugin is not able to open databases when targeting the SDK 30 and running on devices with Android 11.
The issue was already reported storesafe/cordova-sqlite-storage-dependencies#3 and storesafe#954.
From what I was able to understand, it is related with the permissions changes on Android 11, where the SQLite is unable to open the database file using native C primitive `sqlite3_open_v2`:
- https://github.com/liteglue/Android-sqlite-connector/blob/master/src/io/liteglue/SQLGDatabaseHandle.java#L14
- https://github.com/liteglue/Android-sqlite-connector/blob/master/src/io/liteglue/SQLiteNative.java#L60
- https://github.com/liteglue/Android-sqlite-native-driver/blob/master/native/sqlc.c#L40

Causing the handler to crash with a NullPointerException when throwing an exception:
- https://github.com/liteglue/Android-sqlite-connector/blob/master/src/io/liteglue/SQLiteGlueConnection.java#L12

Summing up, this hotfix will enable the usage of the default Android database provider for Android 11 only, by setting the `oldImpl` to `true`, which is the same behaviour of using the `androidDatabaseProvider: 'system'` setting at the Javascript side.

### Context
Fixes: https://outsystemsrd.atlassian.net/browse/RNMT-4515

### CODE REVIEW

File: src/android/io/sqlc/SQLitePlugin.java [hunk 1]
```diff
@@ -337,6 +337,11 @@ private static String getDBConnectionName(String dbname, JSONObject options) {
 
             this.q = new LinkedBlockingQueue<DBQuery>();
             this.openCbc = cbc;
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                Log.v(SQLitePlugin.class.getSimpleName(), "Applying hotfix for Android 11+");
+                this.oldImpl = true;
```
Comment (cmfsotelo) [inline src/android/io/sqlc/SQLitePlugin.java:343]: Can we have an explanation on how this affects the resolution? maybe a github issue link or something on the commit message?

File: CHANGES.md [hunk 2]
```diff
@@ -1,4 +1,7 @@
 # Changes
+### Unreleased
```
Comment (cmfsotelo) [inline CHANGES.md:2]: Never saw this pattern...
We use this on forks?

**Review by cmfsotelo — COMMENTED** [review]
Minor Question

File: src/android/io/sqlc/SQLitePlugin.java (hunk 1 above)
Comment (usernuno) [reply src/android/io/sqlc/SQLitePlugin.java:343]: We have this one, but it doesn't refer the `oldImpl` field.

https://github.com/storesafe/cordova-sqlite-storage-dependencies/issues/3

**Review by usernuno — APPROVED** [review]
Approving, but I agree with @cmfsotelo. We should have a bit more of context in the PR.

File: src/android/io/sqlc/SQLitePlugin.java (hunk 1 above)
Comment (luissilvaos) [reply src/android/io/sqlc/SQLitePlugin.java:343]: The `this.oldImpl = true`  is the same result of using the mentioned workaround `androidDatabaseProvider: 'system'`.

File: CHANGES.md (hunk 2 above)
Comment (luissilvaos) [reply CHANGES.md:2]: I think not. But we are updating this changelog.

File: src/android/io/sqlc/SQLitePlugin.java (hunk 1 above)
Comment (oscsx) [inline src/android/io/sqlc/SQLitePlugin.java:343]: What does this do? Is it equivalent to use the `androidDatabaseProvider: 'system'` setting? There are some glitches when using that setting: https://github.com/storesafe/cordova-sqlite-storage#android-database-provider

Comment (luissilvaos) [reply src/android/io/sqlc/SQLitePlugin.java:343]: I will add more context to the commit message.

Comment (luissilvaos) [reply src/android/io/sqlc/SQLitePlugin.java:343]: Yes, it is, but it will only be enable for Android 11. Regarding the database lock issue, it seems to no longer be an issue in the most recent Android versions. On the other hand, the data corruption issue appears to be caused by the use of multiple plugins to access the same database.

Comment (oscsx) [reply src/android/io/sqlc/SQLitePlugin.java:343]: The app always uses the same plugin, so it should not be a problem.

Comment (luissilvaos) [reply src/android/io/sqlc/SQLitePlugin.java:343]: Done

File: src/android/io/sqlc/SQLitePlugin.java [hunk 3]
```diff
@@ -337,6 +337,11 @@ private static String getDBConnectionName(String dbname, JSONObject options) {
 
             this.q = new LinkedBlockingQueue<DBQuery>();
             this.openCbc = cbc;
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
```
Comment (cmfsotelo) [inline src/android/io/sqlc/SQLitePlugin.java:341]: hm, using this as such won't make the plugin fail to compile in older versions of mabs, due to not using the SDK 30, thus not knowing the value of Build.VERSION_CODES.R ?

Just a heads up, that this is not "retro compatible", therefore almost renders the if useless.

**Review by cmfsotelo — APPROVED** [review]
Approving now, but a new question arose, something i only saw now

**Review by usernuno — APPROVED** [review]
Great work with the context!

File: src/android/io/sqlc/SQLitePlugin.java (hunk 3 above)
Comment (luissilvaos) [reply src/android/io/sqlc/SQLitePlugin.java:341]: Yeah, that's true, and I wondered if we should protected this with the targetSdk too. However, we already have some changes that are not retro compatible with other MABS versions than MABS 7.

**Comment by luissilvaos** [comment]
Just an additional note, Android R(11) is using SQLite 3.28.0. This version of the plugin is using SQLite 3.26.0.
